File: src/main/java/io/saso/dash/templating/Templater.java
package io.saso.dash.templating;

import com.lyncode.jtwig.JtwigModelMap;
import com.lyncode.jtwig.exception.CompileException;
import com.lyncode.jtwig.exception.ParseException;
import com.lyncode.jtwig.exception.RenderException;

import java.io.IOException;

public interface Templater
{
    String render(String template, JtwigModelMap modelMap) throws IOException,
            ParseException, CompileException, RenderException;
}

File: src/main/kotlin/io/saso/dash/templating/Templater.kt
package io.saso.dash.templating

import com.lyncode.jtwig.JtwigModelMap

public interface Templater
{
    /**
     * Renders a template using the provided model.
     *
     * @param template the name of the template
     * @param modelMap the model to pass to the template
     *
     * @return the rendered template
     */
    fun render(template: String, modelMap: JtwigModelMap): String
}
